pkg/controllers/provisioningv2/rke2/machineprovision/args.go

```diff
@@ -173,7 +164,7 @@ func (h *handler) getArgsEnvAndStatus(infra *infraObject, args map[string]interf
 		Args:                cmd,
 		BackoffLimit:        jobBackoffLimit,
 		RKEMachineStatus: rkev1.RKEMachineStatus{
-			Ready:                     infra.data.String("spec", "providerID") != "" && infra.data.Bool("status", "jobComplete"),
+			Ready:                     infra.data.String("spec", "providerID") != "",
```

Comment:
We reuse job complete -- the same field is used for delete which throws us off.

Reply:
Can we refactor the `jobComplete` to do `createJobComplete` vs `deleteJobComplete`? ready should probably only ever care about the `createJobComplete`, although if the `deleteJobComplete` is being run the ready should be false because it is no longer ready